UOP1_Project/Assets/Scripts/SpawnSystem.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Linq;
+using UnityEditor.PackageManager;
+using UnityEngine;
+using UnityEngine.Events;
+
+[Serializable]
+public class PlayerSpawnEvent : UnityEvent<Transform> { }
+
+public class SpawnSystem : MonoBehaviour
+{
+    private static int requestedSpawnIndex = -1;
+
+    [Header("Settings")]
+    [SerializeField] private int defaultSpawnIndex = 0;
+
+    [Header("Project References")]
+#pragma warning disable 649
+    [SerializeField] private Protagonist playerPrefab;
+#pragma warning restore 649
+
+    [Header("Scene References")]
+    [SerializeField] private InputReader inputReader;
+    [SerializeField] private Transform gameplayCamera;
+    [SerializeField] private Transform[] spawnLocations;
+
+    [Header("Events")]
+    public PlayerSpawnEvent onPlayerSpawnEvent;
+
+    void Awake()
+    {
+        try
```

Comment:
I'ld rather see a full stacktrace in the logs rather than hiding an exception :/

Since this is run in Awake(), it won't prevent any other scripts from running.

Reply:
After reading through the rest of the code, I see this is how you centralized your error management. I'm not exactly sure how I feel about it. While it is nice and tidy, I've never been fond of using exceptions to return "expected" errors. I would personally prefer that the Spawn method return a boolean for success since exceptions can be a bit more destructive.